### ISSUE TITLE
session as sync value

### ISSUE THREAD

**Issue by skokenes** [axisgroup/RxQ#48]
In order to better support things like getting notifications, closing a session and freeing up resources, etc, it could be beneficial to refactor the interface for getting a Session as synchronous. Then you jump into Observable land when interacting with the Session. Enigma.js works like this. API could become:

```javascript
// create an instance of a session
const session = createSession(config);

// to actually interact with the session
const global$ = session.global();

// Notifications from the session
session.notification$

// Close the session
session.close();
```

Some questions may linger about when is the WebSocket connection created, when is it tested for authentication, etc? Does it wait for you to try to leverage the session via `.global()`? Or does it set the session socket, etc up eagerly when you call createSession?

**Comment by skokenes** [comment]
[x] create session instance
[x] update unit tests
[ ] update e2e tests
[ ] implement close logic

**Comment by skokenes** [comment]
merged in v2.0.0 branch
